src/app/auth/forgotPassword/index.tsx

```diff
@@ -1,116 +1,101 @@
 
 function ForgotPasswordScreen() {
   const { updateUser, signOut, resetPassword, verifyOtp } = useSession();
   const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-  const [loading, setLoading] = useState(false);
-  const [verificationCode, setCode] = useState<string>('');
-  const [changingPassword, setChangingPassword] = useState(false);
+  const [emailToReset, setEmailToReset] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [validEmail, setValidEmail] = useState(false);
+
+  const mutex = useRef(new Mutex());
 
   const sendResetEmail = async () => {
-    const { error } = await resetPassword(email);
+    const { error } = await resetPassword(emailToReset);
     if (error)
       Alert.alert('Could not send a reset password email. Please try again.');
-    else
-      Alert.alert(
-        'Enter the verification code from your email to change your password',
-      );
+    else router.replace('auth/signup');
   };
-  const verifyCode = async () => {
-    setLoading(true);
 
-    if (email && verificationCode) {
-      const { error } = await verifyOtp(email, verificationCode);
+  const setAndCheckEmailOrUsername = async (newEmail: string) => {
+    setEmail(newEmail);
+    const release = await mutex.current.acquire();
 
-      if (error) {
-        Alert.alert(error.message);
-        setChangingPassword(false);
+    try {
+      if (validator.isEmail(newEmail)) {
+        if (newEmail.length === 0) {
+          setEmailError('');
+          return;
+        }
+        const emailIsTaken = await isEmailTaken(newEmail);
+        if (!emailIsTaken) {
+          setEmailError(
+            'An account with that email does not exist. Please try again.',
+          );
+          setValidEmail(false);
+          return;
+        }
+        setEmailToReset(newEmail);
       } else {
-        router.push('auth/resetPassword');
-      }
-    } else if (!verificationCode) {
-      Alert.alert(`Please enter a verification code`);
-    } else {
-      Alert.alert(`Please sign up again.`);
-    }
+        const { data, error } = await queryEmailByUsername(newEmail);
 
-    setLoading(false);
-  };
-
-  const changePassword = async () => {
-    setLoading(true);
-    const { error } = await updateUser({ password });
-
-    if (error) {
-      Alert.alert('Updating password failed');
-    } else {
-      await signOut();
-      router.replace('/auth/login');
+        if (data && data?.length > 0 && !error) {
+          setValidEmail(true);
+          setEmailToReset(data[0].email);
+        } else {
+          setEmailError(
+            'An account with that username does not exist. Please try again.',
+          );
+          setValidEmail(false);
+          return;
+        }
+      }
+      setValidEmail(true);
+      setEmailError('');
+    } finally {
+      release();
     }
-
-    setLoading(false);
   };
 
   return (
     <View style={styles.container}>
```

Comment:
I think you forgot to create the "Back" button at the top of the screen from the design. You can pretty much just use the Back button from settings and copy over the styles to implement it. Make sure to double-check the spacings and stuff.